### DISCUSSION TITLE
"Service worker server" for remix

### DISCUSSION THREAD

**Discussion by gut4** [remix-run/remix#5302] [Proposals]
https://twitter.com/ryanflorence/status/1603399586764455936
Browser runtime makes loader/action requests to service worker "server"
and that can talks to indexeddb/replicache/real remix server.

**Comment by cliffordfajardo** [comment]
Currently the remix team is working two new function's `clientLoader` and `clientAction` which would allow easier integration with browser cache's (ex: indexDB, localStorage etc)

### References
- original discussion: https://github.com/remix-run/remix/discussions/4950
- Remix proposal: https://github.com/remix-run/remix/discussions/2853
- Initial implementation work: https://github.com/orgs/remix-run/projects/5/views/1?filterQuery=client&pane=issue&itemId=17883646
